### DISCUSSION TITLE
Child tasks or subtasks

### DISCUSSION THREAD

**Discussion by janbaykara** [procrastinate-org/procrastinate#1253] [Q&A]
Hi. We have some tasks running that spawn batches of other tasks. We've built in a shared UUID into the args of these subtasks in order to track progress (like a loading bar in the user interface) but it's a bit clunky. We're now adding in other metadata for the batch requests, like whether the parent task was triggered by a user or a webhook, and so we're thinking of making a "parent task" Django model that references this UUID.

Is there a first-party approach to parent/child tasks on the roadmap for Procrastinate? Or a best practice for implementing this ourselves?

**Comment by onlyann** [comment]
There is no first party approach to creating workflows in general.

This is an area of interest however, judging from some recently opened issues on that topic.

It would be great to introduce a composable approach that would work not only for basic job sequencing but would also support more dynamic workflows.

For example, I find that https://oban.pro/docs/pro/1.5.0-rc.7/Oban.Pro.Workflow.html solves it elegantly. 

Maybe we can have some of that in Procrastinate.
